src/OrchardCore.Modules/OrchardCore.Liquid/Startup.cs

```diff
@@ -19,29 +21,36 @@ namespace OrchardCore.Liquid
 {
     public class Startup : StartupBase
     {
-        static Startup()
-        {
-            // When accessing a property of a JObject instance
-            TemplateContext.GlobalMemberAccessStrategy.Register<JObject, object>((obj, name) => obj[name]);
-
-            // Prevent JTokens from being converted to an ArrayValue as they implement IEnumerable
-            FluidValue.SetTypeMapping<JObject>(o => new ObjectValue(o));
-            FluidValue.SetTypeMapping<JValue>(o => FluidValue.Create(((JValue)o).Value));
-            FluidValue.SetTypeMapping<System.DateTime>(o => new ObjectValue(o));
-        }
-
         public override void ConfigureServices(IServiceCollection services)
         {
             services.AddScoped<ISlugService, SlugService>();
             services.AddScoped<ILiquidTemplateManager, LiquidTemplateManager>();
 
-            services.AddLiquidFilter<TimeZoneFilter>("local");
-            services.AddLiquidFilter<SlugifyFilter>("slugify");
-            services.AddLiquidFilter<ContentUrlFilter>("href");
-            services.AddLiquidFilter<AbsoluteUrlFilter>("absolute_url");
-            services.AddLiquidFilter<LiquidFilter>("liquid");
-            services.AddLiquidFilter<JsonFilter>("json");
-            services.AddLiquidFilter<JsonParseFilter>("jsonparse");
+            services.Configure<TemplateOptions>(options =>
+            {
+                options.Filters.AddFilter("t", LiquidViewFilters.Localize);
+                options.Filters.AddFilter("html_class", LiquidViewFilters.HtmlClass);
+                options.Filters.AddFilter("shape_properties", LiquidViewFilters.ShapeProperties);
+
+                // When a property of a JObject value is accessed, try to look into its properties
+                options.MemberAccessStrategy.Register<JObject, object>((source, name) => source[name]);
+
+                // Convert JToken to FluidValue
+                options.ValueConverters.Add(x => x is JObject o ? new ObjectValue(o) : null);
+                options.ValueConverters.Add(x => x is JValue v ? v.Value : null);
+                options.ValueConverters.Add(x => x is DateTime d ? new ObjectValue(d) : null);
+
+                options.Filters.AddFilter("json", JsonFilter.Json);
+                options.Filters.AddFilter("jsonparse", JsonParseFilter.JsonParse);
+            })
+            .AddLiquidFilter<TimeZoneFilter>("local")
+            .AddLiquidFilter<SlugifyFilter>("slugify")
```

Comment:
ok, so as we discussed, I've moved all the filters that use DI services to classes, which are instaniated by the `LiquidFilterResolver` as required, with services from the constructor.

I've left all the filters that don't use DI, like the above `JsonParseFilter` as statics.
As that seemed like a good idea ;)